File: analysis/analysers/parsers/DependencyParser/src/test/resources/pipeline/persistence/CreatureEntity.java
package de.sots.cellarsandcentaurs.adapter.persistence;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import java.util.UUID;

@Entity
public class CreatureEntity {
    @Id
    private UUID id;

    public CreatureEntity(UUID id) {
        this.id = id;
    }

    public CreatureEntity() {
    }

    public UUID getId() {
        return id;
    }
}

File: training/kotlin/src/main/kotlin/de/sots/cellarsandcentaurs/adapter/persistence/CreatureEntity.kt
package de.sots.cellarsandcentaurs.adapter.persistence

import java.util.UUID

class CreatureEntity(id: String? = null) {
    val id: String = id ?: UUID.randomUUID().toString()
    var typeName: String = ""
    var walkingSpeed: Int = 0
    var hitPointsMax: Int = 0
}
